client/src/context/cogeneration.tsx

```diff
@@ -0,0 +1,283 @@
+  handleQuestionChosen: (val: number | null) => void;
+};
+
+const CogenerationContext = createContext<CogenerationContextType | undefined>(
+  undefined
+);
+
+const CogenerationProvider = (props: {
+  children?: React.ReactNode;
+}): JSX.Element => {
+  const [generationData, setGenerationData] = useState<GenState>({
+    universalContext: "",
+    jsonOutput: "",
+    logPairs: [],
+    prompts: [],
+    questionStrategy: "verification",
+    questionAnswerPairs: [
+      ["", ""],
+      ["", ""],
+      ["", ""],
+    ],
+    questionChosen: null,
+    showQuestions: false,
+    showDistractors: false,
+    distractors: [""],
+  });
+
+  const handleContextChange = useCallback((val: string) => {
+    setGenerationData((prev) => ({ ...prev, universalContext: val }));
+  }, []);
+
+  const handleDistractorChange = useCallback((val: string, idx: number) => {
+    setGenerationData((prev) => {
+      const newDistractors = [...prev.distractors];
+      newDistractors[idx] = val;
+      return { ...prev, distractors: newDistractors };
+    });
+  }, []);
+
+  const handleRemoveDistractor = useCallback((index: number) => {
+    setGenerationData((prev) => {
+      const newDistractors = prev.distractors.filter((_, idx) => idx !== index);
+      return { ...prev, distractors: newDistractors };
+    });
+  }, []);
+
+  const handleGenerateDistractors = useCallback(
+    (strategy: DistractorStrategy) => {
+      setGenerationData((prev) => {
+        const count =
+          prev.logPairs.filter((entry) => entry.type === "distractor").length +
+          1;
+        const newDistractors = generatedDistractors[strategy];
+        const newLogs = [
+          ...prev.logPairs,
+          {
+            title: `Generate Distractors Call #${count}`,
+            type: "distractor",
+            call: "Call Details",
+            response: "Response Details",
+          },
+        ];
+        const newPrompts = [
+          ...prev.prompts,
+          {
+            title: `Distractor Generation #${count}`,
+            type: "distractor",
+            prompt:
+              "I will provide you with a question and a list of correct answers.\nYou must use this data to generate 'distractors' - incorrect answers that could be plausibly mistaken for correct answers by a human.\nYou must also generate topics relevant topics to each question answer pair,\nYou must also generate what is misundertood for each distractor in order,\nyou must also generate relevant topics for each question and answer pair,\nYou msut also generate related topics for each distractor in order,\nYou must misconceptions (justifcation of why the user would get the answer wrong, should be very short and concise like 5 words max) for each distractor in order\nAt last, you must generate feedback to the each distractor for the question and answer pair, the feedback must not include the correct answer.\nHere is the question: {question}\nHere is the list of answers: {answers}",
```

Comment:
probably a good idea to store these prompts away from the actual code. not a big deal, but it would clean things up.  At the very least encapsulate it in a constant.